libr/core/cmd_anal.c

```diff
@@ -606,120 +607,112 @@ static int anal_fcn_add_bb(RCore *core, const char *input) {
+			// function
+			bitmap[fcn->addr-base_addr] = 'F';
+		}
+	}
+
+
+	// Now we print the list of memory regions that are not assigned to a function
+	chunk_size   = 0;
+	chunk_offset = 0;
+	for(i=0; i<code_size; i++){
+		if (bitmap[i]){
+			// We only print a region is its size is bigger than 15 bytes
+			if (chunk_size >= minlen){
+				fcn = r_anal_get_fcn_in (core->anal, base_addr+chunk_offset, R_ANAL_FCN_TYPE_FCN | R_ANAL_FCN_TYPE_SYM);
+				if (fcn)
+					r_cons_printf ("0x%08"PFMT64x"  %6d   %s\n", base_addr+chunk_offset, chunk_size, fcn->name);
+				else 
+					r_cons_printf ("0x%08"PFMT64x"  %6d\n", base_addr+chunk_offset, chunk_size);
+			}
+			chunk_size = 0;
+			chunk_offset = i+1;
+			continue;
+		}
+		chunk_size+=1;
+	}
+	if (chunk_size >= 16){
+		fcn = r_anal_get_fcn_in (core->anal, base_addr+chunk_offset, R_ANAL_FCN_TYPE_FCN | R_ANAL_FCN_TYPE_SYM);
+		if (fcn)
+			r_cons_printf ("0x%08"PFMT64x"  %6d   %s\n", base_addr+chunk_offset, chunk_size, fcn->name);
+		else 
+			r_cons_printf ("0x%08"PFMT64x"  %6d\n", base_addr+chunk_offset, chunk_size);
+	}
+	free(bitmap);
 }
 
 static void r_core_anal_fmap  (RCore *core, const char *input) {
-    ut64 code_size, base_addr; 
-    RListIter *iter, *iter2;                     
-    RAnalFunction *fcn;
-    RAnalBlock *b;
-    char* bitmap;
-    ut64 i,j;
-    int assigned;
-    int show_color = r_config_get_i (core->config, "scr.color");
-
-    // allocate a buffer for the bitmap
-    base_addr = r_num_get (core->num, "$S");
-    code_size = r_num_get (core->num, "$SS");
-    bitmap = calloc(1,code_size+64); 
-
-    // for each function
-    r_list_foreach (core->anal->fcns, iter, fcn) {
-        // for each basic block in the function
-        r_list_foreach (fcn->bbs, iter2, b) { 
-            // if it is not withing range, continue
-            if ((fcn->addr < base_addr) || (fcn->addr >= base_addr+code_size))
-                continue;
-            // otherwise mark each byte in the BB in the bitmap
-            int counter = 1;
-            for (counter=0; counter < b->size; counter++){
-                 bitmap[b->addr+counter-base_addr] = '='; 
-            }
-            bitmap[fcn->addr-base_addr] = 'F';
-        }
-    }
-    
-    // print the bitmap
-    assigned = 0;
-    for(i=0; i<code_size; i+=64){
-        r_cons_printf ("\n0x%08"PFMT64x"  ", base_addr+i);
-        for(j=0; j<64; j++){
-            if ((i+j)==code_size) break;
-            if (bitmap[i+j]) assigned++;
-            if (show_color){
-                if (bitmap[i+j])
-                    r_cons_printf ("%s%c\x1b[0m", Color_GREEN, bitmap[i+j]);
-                else
-                    r_cons_printf (".");
-            }else
-                r_cons_printf ("%c", bitmap[i+j] ? bitmap[i+j] : '.' );
-        }
-    }
-    r_cons_printf ("\n%d / %d (%.2lf%%) bytes assigned to a function\n", assigned, code_size, 100.0*((float)assigned)/code_size);
-    free(bitmap);
+	ut64 code_size = r_num_get (core->num, "$SS");
```

Comment:
same as above. this code looks quite copypasta, so i guess we can generalize this info r_print